File: strings/1273.c
#include <stdio.h>

int len(char *n)
{
    int cont = 0;
    while(*n++)
        cont ++;
    return cont;
}

int main()
{
    int op, cont = 1;
    scanf("%d%*c", &op);
    while(op)
    {
        if (cont != 1)
            printf("\n");
        char p[op][50];

        int i, maior = 0;
        for(i = 0; i < op; i++)
        {
            scanf("%s", p[i]);

            int tam = len(p[i]);
            if (tam > maior)
                maior = tam;
        }
        int j;
        for (i = 0; i < op; i++)
        {
            for(j = 0; j < (maior - len(p[i])); j++){
                printf(" ");
            }
            printf("%s\n", p[i]);
        }
        scanf("%d%*c", &op);
        cont ++;
    }

return 0;
}

File: strings/1273.py
def main():
    cont = 0;
    while True:
        n = int(input())
        if n == 0:
            break;
        elif cont != 0:
            print()
        entrada = []
        maior = 0
        for i in range(n):
            entrada1 = input()
            entrada.append(entrada1)

        maior = max([len(i) for i in entrada])

        for i in entrada:
            print('{}{}'.format(' ' * (maior - len(i)), i))
        cont += 1

if __name__ == '__main__':
    main()